### PR TITLE
+ list-component

### PR DESCRIPTION
**Pull request by dominikvoda**

### CODE REVIEW

File: src-php/Typography/Lists/ListComponent.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+     */
+    private $listItems;
+
+    /**
+     * @var ListType|null
+     */
+    private $listType;
+
+
+    public function __construct(array $items, ?ListType $listType = null)
```
Comment (michalbrauner) [inline src-php/Typography/Lists/ListComponent.php]: can it be not nullable? so everyone outside this class can see the `listType` ?

File: src-php/Typography/Lists/ListItem.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+     */
+    private $value;
+
+    /**
+     * @var ListType|null
+     */
+    private $listItemType;
+
+
+    public function __construct(string $value, ?ListItemType $listItemType = null)
```
Comment (michalbrauner) [inline src-php/Typography/Lists/ListItem.php]: not nullable?

File: src-php/Typography/CodeSnippet.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?php declare(strict_types = 1);
+
+namespace BrandEmbassy\Components\Typography;
+
+use BrandEmbassy\Components\UiComponent;
+
+final class CodeSnippet implements UiComponent
```
Comment (janveselynet) [inline src-php/Typography/CodeSnippet.php]: I'm confused, why is this in two PRs?

File: src-php/Typography/Lists/ListItem.php [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+     */
+    private $value;
+
+    /**
+     * @var ListItemType
+     */
+    private $listItemType;
+
+
+    public function __construct(string $value, ListItemType $listItemType)
```
Comment (janveselynet) [inline src-php/Typography/Lists/ListItem.php]: Maybe value can also be UiComponent, not only string?

File: src-php/Typography/Lists/ListComponent.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    private $listItems;
+
+    /**
+     * @var ListType
+     */
+    private $listType;
+
+
+    /**
+     * @param ListItem[] $items
```
Comment (janveselynet) [inline src-php/Typography/Lists/ListComponent.php]: Missing second parameter

**Pull request merged**